### ISSUE TITLE
[Accessibility]: Improve wording for (n displayed) in ref panel

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#36580]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33510

### Problem description

When navigating the ref panel, there is a (n displayed) next to Definitions and References. However, from the POV of a screen reader user, it is not clear what this means (are there more which are hidden? is that the repo count? is that the number of occurrences?).

### Expected behavior

Change wording to announce number of occurrences in total, that seems like the most useful thing. If some occurrences are not displayed, the screen reader should say that too.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
